tests/components/sonos/test_media_browser.py

```diff
@@ -95,3 +97,147 @@ async def test_build_item_response(
         browse_item.children[1].media_content_id
         == "x-file-cifs://192.168.42.10/music/The%20Beatles/Abbey%20Road/03%20Something.mp3"
     )
+
+
+async def test_browse_media_root(
+    hass: HomeAssistant,
+    soco_factory: SoCoMockFactory,
+    async_autosetup_sonos,
+    soco,
+    discover,
+    hass_ws_client: WebSocketGenerator,
+) -> None:
+    """Test the async_browse_media method."""
+
+    client = await hass_ws_client()
+    await client.send_json(
+        {
+            "id": 1,
+            "type": "media_player/browse_media",
+            "entity_id": "media_player.zone_a",
+        }
+    )
+    response = await client.receive_json()
+    assert response["success"]
+    assert len(response["result"]["children"]) == 2
+    favorites = {
+        "title": "Favorites",
+        "media_class": "directory",
+        "media_content_type": "favorites",
+        "media_content_id": "",
+        "can_play": False,
+        "can_expand": True,
+        "thumbnail": "https://brands.home-assistant.io/_/sonos/logo.png",
+        "children_media_class": None,
+    }
+    assert favorites in response["result"]["children"]
+
+    music_library = {
+        "title": "Music Library",
+        "media_class": "directory",
+        "media_content_type": "library",
+        "media_content_id": "",
+        "can_play": False,
+        "can_expand": True,
+        "thumbnail": "https://brands.home-assistant.io/_/sonos/logo.png",
+        "children_media_class": None,
+    }
+    assert music_library in response["result"]["children"]
+
+
+async def test_browse_media_library(
+    hass: HomeAssistant,
+    soco_factory: SoCoMockFactory,
+    async_autosetup_sonos,
+    soco,
+    discover,
+    hass_ws_client: WebSocketGenerator,
+) -> None:
+    """Test the async_browse_media method."""
+
+    client = await hass_ws_client()
+    await client.send_json(
+        {
+            "id": 1,
+            "type": "media_player/browse_media",
+            "entity_id": "media_player.zone_a",
+            "media_content_id": "",
+            "media_content_type": "library",
+        }
+    )
+    response = await client.receive_json()
+    assert response["success"]
+    assert len(response["result"]["children"]) == 7
```

Comment:
Why do you only assert 3 out of 7 result items?

Reply:
That looks like laziness on my part :-(.  I've added all 7; and updated the next test which had a similar deficiency